src/gpu/intel/sycl/engine.hpp

```diff
@@ -150,12 +150,8 @@ class engine_t : public gpu::intel::compute::compute_engine_t {
             return status::invalid_arguments;
         }
 
-        std::unique_ptr<gpu::intel::ocl::engine_t, engine_deleter_t> ocl_engine;
-        CHECK(gpu::intel::sycl::create_ocl_engine(&ocl_engine, this));
-
         auto kernel_name = jitter->kernel_name();
-
-        xpu::binary_t kernel_binary = jitter->get_binary(ocl_engine.get());
+        xpu::binary_t kernel_binary = jitter->get_binary(this);
```

Comment:
Would it make sense to just use `getKernel` rather than implement all the binary -> runtime object logic in oneDNN?

Reply:
That makes sense, thanks. I did some refactoring in the last commit - please take a look.